src/main/java/fi/otavanopisto/kuntaapi/server/integrations/linkedevents/updaters/LinkedEvenstEventEntityUpdater.java

```diff
@@ -0,0 +1,139 @@
+import fi.otavanopisto.kuntaapi.server.tasks.IdTask.Operation;
+
+@ApplicationScoped
+@Singleton
+@AccessTimeout (unit = TimeUnit.HOURS, value = 1l)
+@SuppressWarnings ("squid:S3306")
+public class LinkedEvenstEventEntityUpdater extends EntityUpdater {
+
+  @Inject
+  private Logger logger;
+
+  @Inject
+  private OrganizationSettingController organizationSettingController; 
+  
+  @Inject
+  private LinkedEventsTranslator linkedEventsTranslator;
+  
+  @Inject
+  private LinkedEventsIdFactory linkedEventsIdFactory;
+
+  @Inject
+  private KuntaApiIdFactory kuntaApiIdFactory;
+  
+  @Inject
+  private LinkedEventsApi linkedEventsApi;
+  
+  @Inject
+  private IdentifierController identifierController;
+
+  @Inject
+  private IdentifierRelationController identifierRelationController;
+
+  @Inject
+  private LinkedEventsEventResourceContainer linkedEventsEventCache;
+  
+  @Inject
+  private ModificationHashCache modificationHashCache;
+
+  @Inject
+  private LinkedEventsEventIdTaskQueue linkedEventsEventIdTaskQueue;
+
+  @Override
+  public String getName() {
+    return "linkedevents-events";
+  }
+
+  @Override
+  public void timeout() {
+    executeNextTask();
+  }
+  
+  private void executeNextTask() {
+    IdTask<EventId> task = linkedEventsEventIdTaskQueue.next();
+    if (task != null) {
+      if (task.getOperation() == Operation.UPDATE) {
+        updateLinkedEventsEvent(task.getId(), task.getOrderIndex()); 
+      } else if (task.getOperation() == Operation.REMOVE) {
+        deleteLinkedEventsEvent(task.getId());
+      }
+    }
+  }
+  
+  private void updateLinkedEventsEvent(EventId eventId, Long orderIndex) {
+    OrganizationId organizationId = eventId.getOrganizationId();
+    if (!organizationSettingController.hasSettingValue(organizationId, LinkedEventsConsts.ORGANIZATION_SETTING_BASEURL)) {
+      logger.log(Level.INFO, "Organization linkedEvents baseUrl not set, skipping update"); 
+      return;
+    }
+    
+    EventApi eventApi = linkedEventsApi.getEventApi(organizationId);
+    ApiResponse<Event> response = eventApi.eventRetrieve(eventId.getId());
+    if (response.isOk()) {
+      updateLinkedEventsEvent(organizationId, response.getResponse(), orderIndex);
+    } else {
+      logger.warning(String.format("Find event %s event %s failed on [%d] %s", organizationId.getId(), eventId.toString(), response.getStatus(), response.getMessage()));
+    }
+  }
+  
+  private void updateLinkedEventsEvent(OrganizationId organizationId, Event linkedEventsEvent, Long orderIndex) {
+    EventId linkedEventsEventId = linkedEventsIdFactory.createEventId(organizationId, linkedEventsEvent.getId());
+        
+    Identifier identifier = identifierController.acquireIdentifier(orderIndex, linkedEventsEventId);
+    identifierRelationController.setParentId(identifier, organizationId);
+    
+    EventId kuntaApiEventId = kuntaApiIdFactory.createFromIdentifier(EventId.class, identifier);
+    // TODO: Load place
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)